libs/ledger-live-common/src/hw/getOnboardingStatePolling.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  from,
+  merge,
+  partition,
+  of,
+  throwError,
+  Observable,
+  TimeoutError,
+} from "rxjs";
+import { map, catchError, repeat, first, timeout } from "rxjs/operators";
+import getVersion from "./getVersion";
+import { withDevice } from "./deviceAccess";
+import {
+  TransportStatusError,
+  DeviceOnboardingStatePollingError,
+  DeviceExtractOnboardingStateError,
+  DisconnectedDevice,
+  CantOpenDevice,
+} from "@ledgerhq/errors";
+import { FirmwareInfo } from "../types/manager";
+import {
+  extractOnboardingState,
+  OnboardingState,
+} from "./extractOnboardingState";
+
+export type OnboardingStatePollingResult = {
+  onboardingState: OnboardingState | null;
+  allowedError: Error | null;
+};
+
+/**
+ * Polls the device onboarding state at a given frequency
+ * @param deviceId A device id
+ * @param pollingPeriodMs The period in ms after which the device onboarding state is fetched again
+ * @param fetchingTimeoutMs The time to wait while fetching for the device onboarding state before throwing an error, in ms
+ * @returns An Observable that polls the device onboarding state
+ */
+export const getOnboardingStatePolling = ({
+  deviceId,
+  pollingPeriodMs,
+  fetchingTimeoutMs = pollingPeriodMs,
+}: {
+  deviceId: string;
+  pollingPeriodMs: number;
+  fetchingTimeoutMs?: number;
+}): Observable<OnboardingStatePollingResult> => {
+  let firstRun = true;
+
+  const delayedOnboardingStateOnce$: Observable<OnboardingStatePollingResult> =
+    new Observable((subscriber) => {
+      const delayMs = firstRun ? 0 : pollingPeriodMs;
+      firstRun = false;
+
+      const getOnboardingStateOnce = () => {
+        const firmwareInfoOrAllowedError$ = withDevice(deviceId)((t) =>
```

Comment:
what is this naming madness, giving me jquery vibes

Reply:
Ahaha 
I was trying to find a convention to say that a variable was an Observable to be able to read and understand the code quickly (with TS you have the type, but it also infers implicitly a lot, so you don't see directly the type). And according [to the angular doc on rxjs](https://angular.io/guide/rx-library#naming-conventions-for-observables) the $ suffix is a convention that some people follows.

I can easily remove it ;) but i feel the Observable variable names needs something so it explicitly says it's not directly the value but the Observable (same for a stream) <- i could just add a "Obs" suffix ?